test/unittests.cpp

```diff
@@ -891,40 +1254,53 @@ void TestG2MultiExpExpected()
     };
     vector<g2> bases = {one, one};
     g2 expected, result;
-    expected = one.mulScalar<1>({5});
-    result = g2::multiExp(bases, scalars).value();
+    expected = one.scale<1>({5});
+    result = g2::weightedSum(bases, scalars);
     if(!expected.equal(result))
     {
         throw invalid_argument("bad multi-exponentiation");
     }
 }
 
-void TestG2MultiExpBatch()
+void TestG2WeightedSumBatch()
 {
-    g2 one = g2::one();
-    int64_t n = 1000;
-    vector<array<uint64_t, 4>> scalars;
-    vector<g2> bases;
-    // scalars: [s0,s1 ... s(n-1)]
-    // bases: [P0,P1,..P(n-1)] = [s(n-1)*G, s(n-2)*G ... s0*G]
-    for(int64_t i = 0, j = n-1; i < n; i++, j--)
-    {
-        scalars.insert(scalars.begin(), array<uint64_t, 4>{static_cast<uint64_t>(rand()%100000), 0, 0, 0});
-        bases.push_back(g2::zero());
-        bases[i] = one.mulScalar(scalars[0]);
-    }
-    // expected: s(n-1)*P0 + s(n-2)*P1 + s0*P(n-1)
-    g2 expected, tmp;
-    for(int64_t i = 0; i < n; i++)
-    {
-        tmp = bases[i].mulScalar(scalars[i]);
-        expected = expected.add(tmp);
-    }
-    g2 result = g2::multiExp(bases, scalars).value();
-    if(!expected.equal(result))
-    {
-        throw invalid_argument("bad multi-exponentiation");
-    }
+    const auto doTest = [](int64_t n) {
+        g2 one = g2::one();
+        vector<array<uint64_t, 4>> scalars;
+        vector<g2> bases;
+        // scalars: [s0,s1 ... s(n-1)]
+        // bases: [P0,P1,..P(n-1)] = [s(n-1)*G, s(n-2)*G ... s0*G]
+        for(int64_t i = 0; i < n; i++)
+        {
+            scalars.push_back(random_scalar());
+            bases.push_back(random_g2());
+        }
+        // expected: s(n-1)*P0 + s(n-2)*P1 + s0*P(n-1)
+        g2 expected, tmp;
+        for(int64_t i = 0; i < n; i++)
+        {
+            tmp = bases[i].scale(scalars[i]);
+            expected = expected.add(tmp);
+        }
+        g2 result = g2::weightedSum(bases, scalars);
+        if(!expected.equal(result))
+        {
+            throw invalid_argument("bad multi-exponentiation");
```

Comment:
Error message should say `"bad g2 weighted sum"`.

Also, the comments in this function are misleading.

Reply:
fixed